playwright/_impl/_fetch.py

```diff
@@ -429,6 +430,23 @@ async def storage_state(
         return result
 
 
+def params_to_protocol(params: Optional[ParamsType]) -> Optional[List[NameValue]]:
+    if not params:
+        return None
+    if isinstance(params, dict):
+        return object_to_array(params)
+    if params.startswith("?"):
+        params = params[1:]
+    parsed = parse_qs(params)
+    if not parsed:
+        return None
+    out = []
+    for name, values in parsed.items():
```

Comment:
I wonder why don't we pass a string over the protocol instead? That would be aligned with our approach to do as much on the server, to avoid aligning the details on the client side. Let's discuss with the team.